quesma/model/pipeline_aggregations/bucket_script.go

```diff
@@ -4,44 +4,85 @@ package pipeline_aggregations
-	return BucketScript{ctx: ctx}
+func NewBucketScript(ctx context.Context, script string) BucketScript {
+	return BucketScript{script: script, PipelineAggregation: newPipelineAggregation(ctx, "_count")}
 }
 
 func (query BucketScript) AggregationType() model.AggregationType {
 	return model.PipelineMetricsAggregation // not sure
 }
 
 func (query BucketScript) TranslateSqlResponseToJson(rows []model.QueryResultRow) model.JsonMap {
-	if len(rows) == 0 {
-		logger.WarnWithCtx(query.ctx).Msg("no rows returned for bucket script aggregation")
-		return model.JsonMap{"value": 0}
-	}
-	var response []model.JsonMap
-	for _, row := range rows {
-		response = append(response, model.JsonMap{"value": row.Cols[0].Value})
-	}
-	return model.JsonMap{
-		"buckets": response,
+	const defaultValue = 0.
+	switch {
+	case query.script == "params.numerator != null && params.denominator != null && params.denominator != 0 ? params.numerator / params.denominator : 0":
+		numerator := query.findFilterValue(rows, "numerator")
+		denominator := query.findFilterValue(rows, "denominator")
+		if denominator == 0 {
+			return model.JsonMap{"value": defaultValue}
+		}
+		return model.JsonMap{"value": numerator / denominator}
+	case len(rows) == 1:
+		for _, row := range rows {
+			return model.JsonMap{"value": util.ExtractNumeric64(row.LastColValue())}
+		}
 	}
+
+	logger.WarnWithCtx(query.ctx).Msgf("unexpected result in bucket_script: %s, len(rows): %d. Returning default.", query.String(), len(rows))
+	return model.JsonMap{"value": defaultValue}
 }
 
-func (query BucketScript) CalculateResultWhenMissing(*model.Query, []model.QueryResultRow) []model.QueryResultRow {
-	return []model.QueryResultRow{}
+func (query BucketScript) CalculateResultWhenMissing(parentRows []model.QueryResultRow) []model.QueryResultRow {
+	resultRows := make([]model.QueryResultRow, 0, len(parentRows))
+	for _, parentRow := range parentRows {
+		resultRow := parentRow.Copy()
+		if len(resultRow.Cols) != 0 {
+			resultRow.Cols[len(resultRow.Cols)-1].Value = util.ExtractNumeric64(parentRow.LastColValue())
+		} else {
+			logger.ErrorWithCtx(query.ctx).Msgf("unexpected empty parent row in bucket_script: %s", query.String())
+		}
+		resultRows = append(resultRows, resultRow)
+	}
+	return resultRows
 }
 
 func (query BucketScript) String() string {
-	return "bucket script"
+	return fmt.Sprintf("bucket_script(isCount: %v, parent: %s, pathToParent: %v, parentBucketAggregation: %v, script: %v)",
+		query.isCount, query.Parent, query.PathToParent, query.parentBucketAggregation, query.script)
 }
 
 func (query BucketScript) PipelineAggregationType() model.PipelineAggregationType {
-	return model.PipelineParentAggregation // not sure, maybe it's sibling.
+	return model.PipelineParentAggregation // not sure, maybe it's sibling. change hasn't changed the result when running some tests.
+}
+
+func (query BucketScript) findFilterValue(rows []model.QueryResultRow, filterName string) float64 {
+	const defaultValue = 0.0
```

Comment:
Does not like this logic, during query generation we should save exact filter name and reuse it here.

Reply:
I don't understand, isn't that exactly what I'm doing? Passing `filterName` as an argument.
I don't save it now, as handling only `numerator/denominator` works